E3Next/Processors/GiveItems.cs

```diff
@@ -0,0 +1,339 @@
+                if (SlotExists)
+                {
+                    Int32 ContainerSlots = MQ.Query<Int32>($"${{Me.Inventory[pack{i}].Container}}");
+
+                    if (ContainerSlots > 0)
+                    {
+                        for (Int32 e = 1; e <= ContainerSlots; e++)
+                        {
+                            String bagItem = MQ.Query<String>($"${{Me.Inventory[pack{i}].Item[{e}]}}");
+                            if (allItemNames.Contains(bagItem))
+                            {
+                                InventoryPosition position = new InventoryPosition(pack: i, slot: e);
+                                InventoryItem item = new InventoryItem(position:position, item:bagItem);
+                                itemsList.Add(item);
+                            }
+                        }
+                    }
+                }
+            }
+            return itemsList;
+        } 
+
+        /// Doing an extra pass through, but Big O tells me it's okay.
+        /// I convert the inventory item to a dictionary with the itemName as the key, and the value for all items for that name.
+        /// This deals with having multiple of the same item in your inventory for giving purposes.
+        private static void HashInventory(List<InventoryItem> inventory)
+        {
+            itemsDict = new Dictionary<string, List<InventoryItem>>();
+            foreach (InventoryItem item in inventory) {
+                List<InventoryItem> itemsForKey;
+                if (!itemsDict.TryGetValue(item.item, out itemsForKey)) {
+                    itemsForKey = new List<InventoryItem>();
+                }
+                itemsForKey.Add(item);
+                itemsDict.Add(item.item, itemsForKey);
+            }
+        }
+        #endregion
+
+        #region Reading Give Items from INI
+        /// If we're passed in a category, only load that category. Otherwise, load all categories
+        private static void LoadGiveItems(string category="")
+        {
+            parsedGiveItems = new Dictionary<string, List<string>>();
+            IniData parsedData = ReadSettingsIni();
+
+            allItemNames = new List<string>();
+            var enumerator = parsedData.Sections.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                if (category.Length > 0) {
+                    if (enumerator.Current.SectionName.Equals(category)) {
+                        LoadSettingsINISection(enumerator.Current);
+                        break;
+                    }
+                } else {
+                    LoadSettingsINISection(enumerator.Current);
+                }
+            }
+            List<InventoryItem> inventory = LoadInventory();
+            HashInventory(inventory);
+        }
+
+        #endregion
+
+        #region Bot INI
+        /// Load the Bot INI to see what they want to accept
+        private static List<string> LoadBotINIForCategories(string botName)
```

Comment:
Could probably change this to just read from GiveItemSettings, vs having this defined in char inis